### PR TITLE
airbyte-ci: set user to airbyte on connector build

### PR DESCRIPTION
**Pull request by alafanechere**
This PRs changes the connector build logic to set the user to `airbyte`.

### CODE REVIEW

**Comment by alafanechere** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36545?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#36545** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36545?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#36544** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36544?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#36537** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36537?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py [hunk 1]
```diff
@@ -172,4 +176,10 @@ async def with_airbyte_java_connector(context: ConnectorContext, connector_java_
         .with_label("io.airbyte.name", context.metadata["dockerRepository"])
         .with_entrypoint(entrypoint)
     )
-    return await finalize_build(context, connector_container)
+    connector_container = await finalize_build(context, connector_container)
+    connector_container = (
+        connector_container.with_exec(["chown", "-R", user, "/airbyte"], skip_entrypoint=True)
+        .with_exec(["chmod", "777", "/tmp"], skip_entrypoint=True)
+        .with_user(user)
+    )
```
Comment (evantahler) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py:184]: I'm not entirely sure why this is needed (shouldn't every user have r/w access to /tmp?), but it is.

**Review by evantahler — APPROVED** [review]

File: airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py:184]: I thought too, but I faced permission denied errors while writing to /tmp during the integration test run on S3

**Review by natikgadzhi — APPROVED** [review]
I am _assuming_ that @bgroff is correct that platform will be able to run rootless images and there are no backward compatibility concerns.

**Pull request merged**